File: src/main/kotlin/cash/andrew/mntrailconditions/ui/MainActivity.java
package cash.andrew.mntrailconditions.ui;

import android.app.Activity;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.ViewGroup;
import butterknife.BindColor;
import butterknife.BindView;
import butterknife.ButterKnife;
import cash.andrew.mntrailconditions.R;
import com.kobakei.ratethisapp.RateThisApp;
import javax.inject.Inject;

public final class MainActivity extends Activity {
  @BindView(R.id.main_content)
  ViewGroup content;

  @BindColor(R.color.status_bar)
  int statusBarColor;

  @Inject ViewContainer viewContainer;

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    LayoutInflater inflater = getLayoutInflater();

    // todo inject stuff

    ViewGroup container = viewContainer.forActivity(this);

    inflater.inflate(R.layout.main_activity, container);
    ButterKnife.bind(this, container);

    inflater.inflate(R.layout.trail_list_view, content);

    RateThisApp.onCreate(this);
    RateThisApp.showRateDialogIfNeeded(this);
  }
}

File: src/main/kotlin/cash/andrew/mntrailconditions/ui/MainActivity.kt
package cash.andrew.mntrailconditions.ui

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.navigation.fragment.NavHostFragment
import androidx.navigation.ui.setupWithNavController
import cash.andrew.mntrailconditions.R
import cash.andrew.mntrailconditions.util.ComponentContainer
import cash.andrew.mntrailconditions.util.DeviceWakeUp
import cash.andrew.mntrailconditions.util.makeComponent
import kotlinx.android.synthetic.main.main_activity.*
import javax.inject.Inject

class MainActivity : AppCompatActivity(), ComponentContainer<ActivityComponent> {

    private inline val bottomNavView get() = bottom_nav_view

    @Inject lateinit var deviceWakeUp: DeviceWakeUp

    override val component: ActivityComponent by lazy { makeComponent() }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.main_activity)

        component.inject(this)
        deviceWakeUp.riseAndShine()

        val navHostFragment = supportFragmentManager
            .findFragmentById(R.id.nav_host_fragment) as NavHostFragment

        bottomNavView.setupWithNavController(navHostFragment.navController)
    }
}
